libvast/src/system/spawn_disk_monitor.cpp

```diff
@@ -22,31 +22,42 @@ spawn_disk_monitor(system::node_actor* self, spawn_arguments& args) {
     return make_error(ec::missing_component, "index");
   if (!archive)
     return make_error(ec::missing_component, "archive");
-  auto hiwater_str
-    = caf::get_or(args.inv.options, "vast.start.disk-budget-high", "0KiB");
-  auto lowater_str
-    = caf::get_or(args.inv.options, "vast.start.disk-budget-low", "0KiB");
-  // TODO: This
+  auto& opts = args.inv.options;
+  // TODO: Also allow integer arguments. Currently we insist on a string because
+  // `caf::get_or` below will silently take the default value of 0KiB if the key
+  // is not a string. (e.g. `disk-budget-high: 6T`)
+  if (opts.contains("vast.start.disk-budget-high")
+      && !caf::get_if<std::string>(&opts, "vast.start.disk-budget-high"))
+    return make_error(ec::invalid_argument, "could not parse disk-budget-high "
+                                            "as byte size");
+  if (opts.contains("vast.start.disk-budget-low")
+      && !caf::get_if<std::string>(&opts, "vast.start.disk-budget-low"))
+    return make_error(ec::invalid_argument, "could not parse disk-budget-low "
+                                            "as byte size");
+  auto hiwater_str = caf::get_or(opts, "vast.start.disk-budget-high", "0KiB");
+  auto lowater_str = caf::get_or(opts, "vast.start.disk-budget-low", "0KiB");
+  VAST_WARNING(hiwater_str, lowater_str);
   auto default_seconds
     = std::chrono::seconds{defaults::system::disk_scan_interval}.count();
-  auto interval = caf::get_or(
-    args.inv.options, "vast.start.disk-budget-check-interval", default_seconds);
+  auto interval = caf::get_or(opts, "vast.start.disk-budget-check-interval",
+                              default_seconds);
   uint64_t hiwater, lowater;
   if (!parsers::bytesize(hiwater_str, hiwater))
     return make_error(ec::parse_error,
                       "could not parse disk budget: " + hiwater_str);
   if (!parsers::bytesize(lowater_str, lowater))
     return make_error(ec::parse_error,
                       "could not parse disk budget: " + lowater_str);
+  VAST_WARNING(hiwater, lowater);
```

Comment:
This looks look like debug information and not ready for user consumption without extra details.

Reply:
Whoops.